pkg/operator/config/templates/agent-integrations.libsonnet

```diff
@@ -0,0 +1,135 @@
+// agent-integrations.libsonnet is the entrypoint for rendering a Grafana Agent
+// config file for integrations based on the Operator custom resources.
+//
+// When writing an object, any field will null will be removed from the final
+// YAML. This is useful as we don't want to always translate unfilled values
+// from the custom resources to a field in the YAML.
+//
+// A series of helper methods to convert default values into null (so they can
+// be trimmed) are in ./ext/optionals.libsonnet.
+//
+// When writing a new function, please document the expected types of the
+// arguments.
+
+local marshal = import 'ext/marshal.libsonnet';
+local optionals = import 'ext/optionals.libsonnet';
+
+local new_integration = import './integrations.libsonnet';
+local new_logs_instance = import './logs.libsonnet';
+local new_metrics_instance = import './metrics.libsonnet';
+local new_external_labels = import 'component/metrics/external_labels.libsonnet';
+local new_remote_write = import 'component/metrics/remote_write.libsonnet';
+
+local calculateShards(requested) =
+  if requested == null then 1
+  else if requested > 1 then requested
+  else 1;
+
```

Comment:
This is a little confusing. Is this because the default is zero? I'd kinda prefer to explode here if somehow you pass a negative. But maybe this is a deep enough utility this behavior is safe enough?

Reply:
Ah, right, I copied this from the existing metrics code since `agent-integrations.libsonnet` is a combination of metrics + logs + integrations. 

We do the logic here because `0` is the default, and we want at least 1 shard. We could probably do a better job at validating input / applying the proper defaults before we pass things over to the jsonnet generator, though.